### PR TITLE
MoEngage iOS v2 device mode updates

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Added steps for MoEngage iOS v2 device mode implementation and push notifications support.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

## Notion ticket link

> [MoEngage device mode updates](https://www.notion.so/rudderstacks/MoEngage-iOS-v2-device-mode-updates-ce613c2c40ce49ae90f0fdf3a010da20)

### CODE REVIEW

**Review by pallabmaiti — APPROVED** [review]

File: docs/destinations/marketing/moengage.mdx [hunk 1]
```diff
@@ -90,6 +91,61 @@ RudderConfigBuilder *builder = [[RudderConfigBuilder alloc] init];
 </ol>
       </TabPanel>
       <TabPanel>
+        <div class="warningBlock">
+          This device mode integration is supported for MoEngage v6.1.0 and above.
+        </div>
+        Follow these steps to add MoEngage to your iOS project:
+<ol>
+<li>Add MoEngage as a destination in the <a href="https://app.rudderstack.com">RudderStack dashboard</a>.</li>
```
Comment (ishachopra0401) [inline docs/destinations/marketing/moengage.mdx:99]: Same as the other ones, removing this step from here and moving it above

File: docs/destinations/marketing/moengage.mdx [hunk 2]
```diff
@@ -538,165 +594,43 @@ RudderStack maps the following properties to the MoEngage properties before send
...
+    </ol>
+      </TabPanel>
+      <TabPanel>
 Copy the Server Key from the FCM console and add it to the MoEngage Dashboard(Not sure where to find the Server Key refer to <a href="https://docs.moengage.com/docs/getting-fcmgcm-server-key">Getting FCM Server Key</a>. To upload it, go to the <a href="https://app.moengage.com/v3/#/settings/push/mobile">Settings Page</a> and add the Server Key and package name.
 **Please make sure you add the keys both in the Test and Live environment.**
-Before configuring MoEngage SDK for receiving push notifications please make you have configured Firebase in your application, if not done already refer to the <a href="https://firebase.google.com/docs/android/setup">documentation</a> and complete setup.
-Also, make sure you have added the Firebase Messaging dependency in your application <code class="inline-code">build.gradle</code> file.
+Before configuring MoEngage SDK for receiving push notifications please make you have configured MoEngage in your application, if not done already refer to the <a href="https://firebase.google.com/docs/android/setup">documentation</a> and complete setup.
```
Comment (ishachopra0401) [inline docs/destinations/marketing/moengage.mdx:632]: I believe the firebase hyperlink should be changed here to MoEngage set up

File: docs/destinations/marketing/moengage.mdx [hunk 3]
```diff
@@ -771,7 +705,7 @@ Add the below code in your manifest file within the application tag
 ```
 </span>
 When MoEngage SDK handles push registration it optionally provides a callback to the Application whenever a new token is registered or token is refreshed.
-An application can get this callback by implementing <code class="inline-code">FirebaseEventListener</code> and registering for a callback in the Application class' <code class="inline-code">onCreate()</code>using <code class="inline-code">MoEFireBaseHelper.getInstance().setEventListener()</code>
+An application can get this callback by implementing <code class="inline-code">MoEngageEventListener</code> and registering for a callback in the Application class' <code class="inline-code">onCreate()</code>using <code class="inline-code">MoEFireBaseHelper.getInstance().setEventListener()</code>
```
Comment (ishachopra0401) [inline docs/destinations/marketing/moengage.mdx:708]: Should this be:
MoEFireBaseHelper.getInstance() -> MoEngageHelper.getInstance()

File: docs/destinations/marketing/moengage.mdx (hunk 1 above)
Comment (ameypv-rudder) [inline docs/destinations/marketing/moengage.mdx:99]:
```suggestion
```

File: docs/destinations/marketing/moengage.mdx (hunk 2 above)
Comment (ameypv-rudder) [inline docs/destinations/marketing/moengage.mdx:632]:
```suggestion
Before configuring MoEngage SDK for receiving push notifications, make you have configured MoEngage in your application.
```

File: docs/destinations/marketing/moengage.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/destinations/marketing/moengage.mdx:708]: @1abhishekpandey @pallabmaiti can you help here?

File: docs/destinations/marketing/moengage.mdx [hunk 4]
```diff
@@ -48,11 +48,12 @@ By turning on the native SDK switch under Android SDK Settings on the RudderStac
   `IND`.
 </div>
 
-## Adding Device Mode Integration
+## Adding device mode integration
```
Comment (ameypv-rudder) [inline docs/destinations/marketing/moengage.mdx:52]:
```suggestion

Once you add MoEngage as a destination in the [RudderStack dashboard](https://app.rudderstack.com/), follow these steps to add it to your project, depending on your integration platform:
```

File: docs/destinations/marketing/moengage.mdx [hunk 5]
```diff
@@ -538,165 +594,43 @@ RudderStack maps the following properties to the MoEngage properties before send
...
+    </ol>
+      </TabPanel>
+      <TabPanel>
 Copy the Server Key from the FCM console and add it to the MoEngage Dashboard(Not sure where to find the Server Key refer to <a href="https://docs.moengage.com/docs/getting-fcmgcm-server-key">Getting FCM Server Key</a>. To upload it, go to the <a href="https://app.moengage.com/v3/#/settings/push/mobile">Settings Page</a> and add the Server Key and package name.
 **Please make sure you add the keys both in the Test and Live environment.**
-Before configuring MoEngage SDK for receiving push notifications please make you have configured Firebase in your application, if not done already refer to the <a href="https://firebase.google.com/docs/android/setup">documentation</a> and complete setup.
-Also, make sure you have added the Firebase Messaging dependency in your application <code class="inline-code">build.gradle</code> file.
+Before configuring the MoEngage SDK for receiving push notifications, make you have integrated MoEngage with your app.
```
Comment (ameypv-rudder) [inline docs/destinations/marketing/moengage.mdx:632]:
```suggestion
Before configuring the MoEngage SDK for receiving push notifications, make sure you have integrated MoEngage with your app.
```

File: docs/destinations/marketing/moengage.mdx (hunk 3 above)
Comment (1abhishekpandey) [reply docs/destinations/marketing/moengage.mdx:708]: It should be MoEFireBaseHelper.getInstance() only. API reference could be found [here](https://developers.moengage.com/hc/en-us/articles/4403896795540-Changelog).

Comment (ameypv-rudder) [inline docs/destinations/marketing/moengage.mdx:708]:
```suggestion
An application can get this callback by implementing <code class="inline-code">FirebaseEventListener</code> and registering for a callback in the Application class' <code class="inline-code">onCreate()</code>using <code class="inline-code">MoEFireBaseHelper.getInstance().setEventListener()</code>
```

File: docs/destinations/marketing/moengage.mdx [hunk 6]
```diff
@@ -538,165 +594,43 @@ RudderStack maps the following properties to the MoEngage properties before send
...
         <div class="infoBlock">
         For more detailed steps please follow <a href="https://docs.moengage.com/docs/push-notification-implementation">MoEngage guidelines here</a>.
         </div>
       <h3>In-App messaging</h3>
       In-app messaging is a type of mobile messaging where the notification is displayed within the app. Some examples include popups, yes/no prompts, interstitials, and more. To implement this please follow this guide: <a href="https://docs.moengage.com/docs/in-app-nativ">InApp NATIV</a>.
       </TabPanel>
       <TabPanel>
+        <ol>
+          <li>In your <code class="inline-code">plist</code> file, add the <code class="inline-code">MoEngageAppDelegateProxyEnabled</code> key and set it <code class="inline-code">false</code> to disable swizzling. </li>
+          <li>Then, generate the <code class="inline-code">.pem</code> file and upload it to MoEngage.</li>
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply docs/destinations/marketing/moengage.mdx]: @1abhishekpandey all this information has been added.

**Review by 1abhishekpandey — APPROVED** [review]

**Pull request merged**